src/mpid/ch4/netmod/ofi/ofi_dynproc.c

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpidimpl.h"
+#include "ofi_impl.h"
+
+#define DISCONNECT_CONTEXT_ID 0xF000
+#define INITIAL_NUM_CONN 10
+
+static int dynproc_send_disconnect(int conn_id);
+static int dynproc_accept_disconnect(int conn_id);
+static int dynproc_grow_conn_table(int new_max);
+static int dynproc_get_next_conn_id(int *conn_id_out);
+
+int MPIDI_OFI_dynproc_init(void)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_CONN_MANAGER_INIT);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_CONN_MANAGER_INIT);
+
+    MPIDI_OFI_global.conn_mgr.n_conn = 0;
+    MPIDI_OFI_global.conn_mgr.max_n_conn = 0;
+    MPIDI_OFI_global.conn_mgr.conn_table = NULL;
+    MPIDI_OFI_global.conn_mgr.n_free = 0;
+    MPIDI_OFI_global.conn_mgr.free_stack = NULL;
+
+  fn_exit:
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_CONN_MANAGER_INIT);
+    return mpi_errno;
+  fn_fail:
+    goto fn_exit;
+}
+
+int MPIDI_OFI_dynproc_finalize(void)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_CONN_MANAGER_DESTROY);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_CONN_MANAGER_DESTROY);
+
+    if (MPIDI_OFI_global.conn_mgr.n_conn > 0) {
+        for (int i = 0; i < MPIDI_OFI_global.conn_mgr.max_n_conn; i++) {
```

Comment:
I think it only needs to iterate up to `MPIDI_OFI_global.conn_mgr.n_conn` -- will fix.